src/main/java/org/mule/module/apikit/uri/TokenOperatorDX.java

```diff
@@ -390,9 +390,6 @@ public static TokenOperatorDX parse(String exp) throws URITemplateSyntaxExceptio
     }
     char c = sexp.charAt(0);
     Operator operator = TokenOperatorDX.toOperator(c);
-    if (operator == null) {
```

Comment:
Are we sure this cannot be reached?

Reply:
yes! 
  public static Operator toOperator(char c) {
    for (Operator o : Operator.values()) {
      if (o.character() == c) {
        return o;
      }
    }
    // default on simple substitution
    return Operator.SUBSTITUTION;
  }